sparta/sparta/resources/SharedData.hpp

```diff
@@ -65,6 +70,44 @@ namespace sparta
             writePS(std::forward<U>(init_val));
         }
 
+        SharedData(const SharedData& rhs) :
+            ev_update_(rhs.ev_update_),
```

Comment:
Why copy the ev_update_ if it's going to be reset?

Reply:
We still need the other fields inside `ev_update_` to get copied.